xonsh/xonfig.py

```diff
@@ -623,6 +637,53 @@ def _web(args):
     subprocess.run([sys.executable, "-m", "xonsh.webconfig"] + args.orig_args[1:])
 
 
+def _kernel(args):
```

Comment:
Can you call this `_jupyter_kernel` please?

Reply:
Ummm, is there a plan to support a kernel for something other than jupyter in the future?  What if jupyter decides to rebrand their project again? why should xonsh be so excruciatingly explicit about the jupyter lernel?

Reply:
Well, so (un)fortunately, because of the way that the amalgamate code works, it squishes all of the source files together to make a single large source file. This means that almost everything shares a single namespace in the xonsh code base. So it is less about other xonfig things being called kernel, and more that we might want to denote other functionality in xonsh as being related to some other kind of kernel (the python executable kernel, the OS kernel, etc).  Amalgamate sort of forces us to be very precise in our naming of everything so that namespace collisions don't happen.

Reply:
I know it is annoying, but it is good to be robust :rofl:

Reply:
OK, I changed the name to `_jupyter_kernel`.  

But what about `_web`, `_wizard` and the other global functions in `xonfig`?  If global uniqueness is really necessary, they could all be renamed `_xonfig_web`, `_xonfig_wizard`, etc, and then the new one should be called `_xonfig_kernel`.  I'm not going there, but I will agree `amalgamate.py` is annoying to live with.

And, see #3790; an additional annoyance is that it doesn't stop the show when it detects name conflicts.